portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -13,26 +11,16 @@
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
-
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        UserService userService = UserServiceFactory.getUserService();
-
-        String userEmail;
-        boolean isLoggedIn = userService.isUserLoggedIn();
-
-        if (isLoggedIn) {
-            userEmail = userService.getCurrentUser().getEmail();
-        } else {
-            userEmail = "";
-        }
-
-        UserData userData = new UserData(userEmail, isLoggedIn);
-
-        Gson gson = new Gson();
-        String responseString = gson.toJson(userData);
-
-        response.setContentType("application/json");
-        response.getWriter().println(responseString);
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Having a "login status" and a "login" servlet seems kind of redundant. It's also weird that there's no POST on here.

Wouldn't it be more elegant to have one Login servlet, where the `POST` does the login, and the `GET` gets the login status?

(not sure if you need to handle logout separately, or if that logoutUrl thing is magic-sauce)

Reply:
Part of my issue is that there's no way to submit a post request using a link. Also, it seems that all of the examples of from the Users API and the walkthrough have the login stuff contained in GET requests. From my understanding, the login service is pretty well separated from all of the requests. If I wanted to extract any user login information, I would end up having to create a separate request. A possible solution would be to create a POST request that takes in whether the user is trying to log in or post data as a query parameter, but I feel like that would also be kind of clunky.

I'm honestly not sure if there is a good way to do this. I initially did it the way I did so that I could reserve the POST request for adding user information to Datastore. If I wanted to do anything with the email addresses themselves, I don't know where I would do that, as that would require a POST request to the servlet somewhere (perhaps on load?).

Reply:
Got it - I think, from looking at the docs a bit, that what actually happens is, the user service just gives you login/logout URLs, and those just direct to the google account setup stuff, so there's nothing to post. I was originally concerned with GET because if you're creating an account or logging in using a GET request, then you'd basically passing passwords on the URL, which would be a terrible practice.